### PR TITLE
Support for Dart & Flutter

### PR DESCRIPTION
**Pull request by jkozal**

### CODE REVIEW

File: src/Parser/DartLintParser.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        return LogSeverity.error;
+      case 'info':
+        return LogSeverity.info;
+      default:
+        return LogSeverity.unknown;
+    }
+  }
+
+  private static isLineValid(lines: string[]): boolean {
+    return lines.length == Object.keys(DartLineEnum).length / 2;
```
Comment (jkozal) [inline src/Parser/DartLintParser.ts:59]: @encX maybe there is a better way how to avoid /2, but im not TS pro :(

**Review by encX — CHANGES_REQUESTED** [review]
Please check comments and also add dart lint to supported section on top. [See doc](https://github.com/codeleague/codecoach/blob/main/CONTRIBUTING.md#development)

File: src/Parser/DartLintParser.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    };
+  }
+
+  private static stringToSeverity(levelText: string): LogSeverity {
+    switch (levelText) {
+      case 'error':
+        return LogSeverity.error;
+      case 'info':
+        return LogSeverity.info;
+      default:
```
[the remark this answers is not in the document]
Comment (jkozal) [reply src/Parser/DartLintParser.ts:55]: has, good catch, i missed them cause dart is not using them a lot ;-)

File: src/Parser/DartLintParser.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import { DartLocation } from './@types/DartLocation';
+import { DartLineEnum } from './@enums/dart.line.enum';
+import { DartLocationLineEnum } from './@enums/dart.location.line.enum';
+
+export class DartLintParser extends Parser {
+  parse(content: string): LogType[] {
+    return content
+      .split(DartIndicatorEnum.newLine)
+      .map((line: string) => DartLintParser.lineToLog(line))
+      .filter((f: LogType | null) => f != null) as LogType[];
```
[the remark this answers is not in the document]
Comment (jkozal) [reply src/Parser/DartLintParser.ts]: done, i handled it with no null

File: src/Parser/DartLintParser.ts (hunk 1 above)
Comment (jkozal) [reply src/Parser/DartLintParser.ts]: done with regexp

File: src/Parser/DartLintParser.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  private static lineToLog(line: string): LogType | null {
+    const lineParts = line.split(DartIndicatorEnum.lineItem);
+    return DartLintParser.isLineValid(lineParts)
+      ? DartLintParser.linePartsToLog(lineParts)
+      : null;
+  }
+
+  private static linePartsToLog(lineParts: string[]): LogType {
+    const location = DartLintParser.linePartsToLocation(
+      lineParts[DartLineEnum.Location].split(DartIndicatorEnum.locationItem),
```
[the remark this answers is not in the document]
Comment (jkozal) [reply src/Parser/DartLintParser.ts]: done with regexp

**Review by encX — APPROVED** [review]

**Pull request merged**